assets/training/distillation/components/data_generation/src/common/utils.py

```diff
@@ -106,43 +113,238 @@ def get_workspace_mlclient(workspace: Workspace = None) -> MLClient:
+
+        Args:
+            mlclient_ws (MLClient): workspace mlclient
+            endpoint_name (str): managed online endpoint name
+
+        Raises:
+            Exception: if fetching endpoint details fail
+            Exception: if endpoint is not in healthy state
+        """
+        self._mlclient: MLClient = mlclient_ws
+        try:
+            self._endpoint: ServerlessEndpoint = self._mlclient.serverless_endpoints.get(endpoint_name)
+        except Exception as e:
+            raise Exception(f"Serverless endpoint fetch details failed with exception: {e}")
+
+        # ensure endpoint is healthy
+        logger.info(f"Endpoint provisioning state: {self._endpoint.provisioning_state}")
+        if not (self._endpoint.provisioning_state.lower() == "succeeded"):
+            raise Exception(f"Endpoint {self._endpoint.name} is unhealthy.")
+
+    def get_endpoint_key(self) -> str:
+        """Get endpoint primary key for serverless deployment.
+
+        Raises:
+            Exception: if fetching key fails
+
+        Returns:
+            str: endpoint primary key for serverless deployment.
+        """
+        try:
+            return self._mlclient.serverless_endpoints.get_keys(self._endpoint.name).primary_key
+        except Exception as e:
+            raise Exception(f"Failed to get endpoint keys for endpoint: {self._endpoint.name}. Exception: {e}")
+
+    def get_endpoint_url(self) -> str:
+        """Get URL for managed online endpoint."""
+        return self._endpoint.scoring_uri
+
+    def get_deployed_model_id(self) -> str:
+        """Return deployed model id."""
+        return self._endpoint.model_id
+
+
+class OnlineEndpointDetails(EndpointDetails):
+    """Online endpoint details for data generation."""
+
+    def __init__(self, mlclient_ws: MLClient, endpoint_name: str):
+        """Initialize endpoint details for managed online endpoint.
+
+        Args:
+            mlclient_ws (MLClient): workspace mlclient
+            endpoint_name (str): managed online endpoint name
+
+        Raises:
+            Exception: if fetching endpoint details fail
+            Exception: if traffic is not allocated or allocated to more than one endpoint
+            Exception: if endpoint or deployment is not in healthy state
+        """
+        self._mlclient: MLClient = mlclient_ws
+        try:
+            self._endpoint: ManagedOnlineEndpoint = self._mlclient.online_endpoints.get(endpoint_name)
+        except Exception as e:
+            raise Exception(f"Online endpoint fetch details failed with exception: {e}")
+
+        all_deployments = self._get_deployments()
+        # fetch deployment with 100% traffic
+        deployments = [
+            deployment
+            for deployment in all_deployments if deployment.name in [
+                deployment_name
+                for deployment_name, traffic_percent in self._endpoint.traffic.items() if traffic_percent == 100
```

Comment:
I am wondering if we want to filter based on the asset(model) uri?

Reply:
we can more than one deployment for a model in an endpoint. How to choose which deployment to hit in that case?